File: 2488-divide-intervals-into-minimum-number-of-groups/2488-divide-intervals-into-minimum-number-of-groups.py
class Solution(object):
    def minGroups(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        start , end = [],[]
        for i,j in intervals:
            start.append(i)
            end.append(j)

        start.sort ()
        end.sort()
        i = 0
        j =0
        ans = 0
        while i< len(intervals):
            if (start[i] <= end[j]):
                i+=1
            else:
                j+=1
            ans = max(ans,i-j)
        return ans

File: 2488-divide-intervals-into-minimum-number-of-groups/2488-divide-intervals-into-minimum-number-of-groups.cpp
class Solution {
public:
    int minGroups(vector<vector<int>>& intervals) {
        vector<int> start ;
        int n = intervals.size();
        vector<int> end;
        for (int i =0;i<n;i++){
            start.push_back(intervals[i][0]);
            end.push_back(intervals[i][1]);
        }
        sort(start.begin(),start.end());
        sort(end.begin(),end.end());
        int grp = 0;
        int i = 0,j =0;
        int ans =0;
        while (i< n){
            if (start[i] <= end[j]){
                grp ++;
                i++;
            }
            else {
                grp --;
                j++;
            }
            ans = max(ans,grp);
        }
        return ans;
    }
};